app/Policies/TrainingPolicy.php

```diff
@@ -109,6 +109,17 @@ public function create(User $user)
         return $user->isModeratorOrAbove();
     }
 
+    /**
+     * Check if the user has access to edit a training details
+     *
+     * @param User $user
+     * @return bool
+     */
+    public function edit(User $user)
+    {
+        return $user->isModeratorOrAbove();
```

Comment:
This should take the area as well, so only a moderator in the given country update the training request.